Modules/Microsoft365DSC/DSCResources/MSFT_IntuneDeviceConfigurationIdentityProtectionPolicyWindows10/MSFT_IntuneDeviceConfigurationIdentityProtectionPolicyWindows10.psm1

```diff
@@ -0,0 +1,1538 @@
+                }
+                if($target.collectionId)
+                {
+                    $formattedTarget.Add('collectionId',$target.collectionId)
+                }
+                if($target.deviceAndAppManagementAssignmentFilterType)
+                {
+                    $formattedTarget.Add('deviceAndAppManagementAssignmentFilterType',$target.deviceAndAppManagementAssignmentFilterType)
+                }
+                if($target.deviceAndAppManagementAssignmentFilterId)
+                {
+                    $formattedTarget.Add('deviceAndAppManagementAssignmentFilterId',$target.deviceAndAppManagementAssignmentFilterId)
+                }
+                $deviceManagementPolicyAssignments+=@{'target'= $formattedTarget}
+            }
+            $body=@{'assignments'=$deviceManagementPolicyAssignments}|ConvertTo-Json -Depth 20
+            #write-verbose -Message $body
+            Invoke-MgGraphRequest -Method POST -Uri $Uri -Body $body -ErrorAction Stop
+
+        }
+        catch
+        {
+            New-M365DSCLogEntry -Message 'Error updating data:'
+                -Exception $_
+                -Source $($MyInvocation.MyCommand.Source)
+                -TenantId $TenantId
+                -Credential $Credential
+
+            return $null
+        }
+
+
+    }function Rename-M365DSCCimInstanceParameter
+{
+    [CmdletBinding()]
+    [OutputType([System.Collections.Hashtable],[System.Collections.Hashtable[]])]
+    param(
+        [Parameter(Mandatory = 'true')]
+        $Properties
+    )
+
+    $keyToRename=@{
+        "odataType"="@odata.type"
+    }
+
+    $result=$Properties
+
+    $type=$Properties.getType().FullName
+
+    #region Array
+    if ($type -like '*[[\]]')
+    {
+        $values = @()
+        foreach ($item in $Properties)
+        {
+            $values += Rename-M365DSCCimInstanceParameter $item
+        }
+        $result=$values
+
+        return ,$result
+    }
+    #endregion
+
+    #region Single
+    if($type -like "*Hashtable")
+    {
+        $result=([Hashtable]$Properties).clone()
+    }
+    if($type -like '*CimInstance*' -or $type -like '*Hashtable*'-or $type -like '*Object*')
+    {
+        $hashProperties = Get-M365DSCDRGComplexTypeToHashtable -ComplexObject $result
+        $keys=($hashProperties.clone()).keys
+        foreach($key in $keys)
+        {
+            $keyName=$key.substring(0,1).tolower()+$key.substring(1,$key.length-1)
+            if ($key -in $keyToRename.Keys)
+            {
+                $keyName=$keyToRename.$key
+            }
+
+            $property=$hashProperties.$key
+            if($null -ne $property)
+            {
+                $hashProperties.Remove($key)
+                $hashProperties.add($keyName,(Rename-M365DSCCimInstanceParameter $property))
+            }
+        }
+        $result = $hashProperties
+    }
+
+    return $result
+    #endregion
+}
+function Get-M365DSCDRGComplexTypeToHashtable
```

Comment:
Please add an empty line between the closing brace and the function

Reply:
done